### PR TITLE
Feature/compliance year

### PR DESCRIPTION
**Pull request by JB12298**

### CODE REVIEW

File: src/EA.Weee.RequestHandlers.Tests.Unit/Admin/AddAatfRequestHandlerTests.cs [hunk 1]
```diff
@@ -78,7 +78,7 @@ public async Task HandleAsync_ValidInput_AddsAatf()
         {
             IWeeeAuthorization authorization = A.Fake<IWeeeAuthorization>();
 
-            AatfData aatf = new AatfData(Guid.NewGuid(), "name", "approval number", A.Dummy<Core.Shared.UKCompetentAuthorityData>(), Core.AatfReturn.AatfStatus.Approved, A.Dummy<AatfAddressData>(), Core.AatfReturn.AatfSize.Large, DateTime.Now);
+            AatfData aatf = new AatfData(Guid.NewGuid(), "name", "approval number", (Int16)2019, A.Dummy<Core.Shared.UKCompetentAuthorityData>(), Core.AatfReturn.AatfStatus.Approved, A.Dummy<AatfAddressData>(), Core.AatfReturn.AatfSize.Large, DateTime.Now);
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.RequestHandlers.Tests.Unit/Admin/AddAatfRequestHandlerTests.cs]: Do a check for the compliance year in the must have happened part below

File: src/EA.Weee.RequestHandlers/Mappings/AatfMap.cs [hunk 2]
```diff
@@ -49,7 +49,7 @@ public AatfData Map(Aatf source)
 
             var organisation = organisationMap.Map(source.Organisation);
 
-            return new AatfData(source.Id, source.Name, source.ApprovalNumber, competentAuthority, aatfStatus, address, aatfSize, source.ApprovalDate.GetValueOrDefault())
+            return new AatfData(source.Id, source.Name, source.ApprovalNumber, source.ComplianceYear, competentAuthority, aatfStatus, address, aatfSize, source.ApprovalDate.GetValueOrDefault())
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.RequestHandlers/Mappings/AatfMap.cs]: Check the compliance year property is set, looks like some other properties have been missed in the unit test

File: src/EA.Weee.RequestHandlers.Tests.Unit/Admin/AddAatfRequestHandlerTests.cs (hunk 1 above)
Comment (JB12298) [reply src/EA.Weee.RequestHandlers.Tests.Unit/Admin/AddAatfRequestHandlerTests.cs]: Added

File: src/EA.Weee.Domain.Tests.Unit/AatfReturn/AatfTests.cs [hunk 3]
```diff
@@ -89,7 +89,7 @@ public void Aatf_GivenValidParameters_AatfPropertiesShouldBeSet()
             var aatfStatus = AatfStatus.Approved;
             var contact = A.Fake<AatfContact>();
 
-            var aatf = new Aatf(name, competentAuthority, approvalNumber, aatfStatus, organisation, A.Fake<AatfAddress>(), A.Fake<AatfSize>(), DateTime.Now, A.Fake<AatfContact>(), A.Dummy<FacilityType>());
+            var aatf = new Aatf(name, competentAuthority, approvalNumber, aatfStatus, organisation, A.Fake<AatfAddress>(), A.Fake<AatfSize>(), DateTime.Now, A.Fake<AatfContact>(), A.Dummy<FacilityType>(), A.Dummy<Int16>());
```
[the remark this answers is not in the document]
Comment (JB12298) [reply src/EA.Weee.Domain.Tests.Unit/AatfReturn/AatfTests.cs]: Added

File: src/EA.Weee.RequestHandlers/Mappings/AatfMap.cs (hunk 2 above)
Comment (JB12298) [reply src/EA.Weee.RequestHandlers/Mappings/AatfMap.cs]: Added checks for the remaining props

File: src/EA.Weee.RequestHandlers/AatfReturn/Internal/EditAatfDetailsRequestHandler.cs [hunk 4]
```diff
@@ -52,7 +52,8 @@ public async Task<bool> HandleAsync(EditAatfDetails message)
                 Enumeration.FromValue<Domain.AatfReturn.AatfSize>(message.Data.Size.Value),
                 message.Data.ApprovalDate.GetValueOrDefault(),
                 existingAatf.Contact,
-                existingAatf.FacilityType);
+                existingAatf.FacilityType,
+                existingAatf.ComplianceYear);
```
[the remark this answers is not in the document]
Comment (JB12298) [reply src/EA.Weee.RequestHandlers/AatfReturn/Internal/EditAatfDetailsRequestHandler.cs]: Done

File: src/EA.Weee.Domain/AatfReturn/Aatf.cs [hunk 5]
```diff
@@ -30,11 +30,23 @@ public class Aatf : Entity
...
         public virtual FacilityType FacilityType { get; private set; }
 
+        public virtual Int16 ComplianceYear { get; private set; }
+
         public Aatf()
         {
         }
 
-        public Aatf(string name, UKCompetentAuthority competentAuthority, string approvalNumber, AatfStatus aatfStatus, Organisation organisation, AatfAddress aatfSiteAddress, AatfSize aatfSize, DateTime approvalDate, AatfContact contact, FacilityType facilityType)
+        public Aatf(string name,
```
[the remark this answers is not in the document]
Comment (JB12298) [reply src/EA.Weee.Domain/AatfReturn/Aatf.cs]: Leaving as it's too much work

**Pull request merged**